qutip/cy/spmath.pyx

```diff
@@ -695,8 +695,9 @@ cdef bint _zcsr_isherm_full(
         free_CSR(&transpose)
 
 
+# Exception signal value must not be something we will ever want to return.
 @cython.overflowcheck(True)
-cdef _safe_multiply(int A, int B):
+cdef int _safe_multiply(int A, int B) except -1:
```

Comment:
Using -1 means we can never use it for negative values. Since it's used to compute sizes it's not an issue, but why not use `except? -1`. It should not affect performance and since it's not clear otherwise that this function would not accept negative values, it feels safer.
Otherwise everything seems fine.

Reply:
Great point - I'd got so used to using `except *` with objects and complex numbers I forgot `except?` existed.  This is absolutely the right way to do it.